### PR TITLE
Refactor Exposure classes and adjust tests

### LINKED ISSUE #305: [✨ Feature request]: ABC Exposure class
### Description

Exposure classes are now defined in a specific version of the Eurocode. The description of the exposure classes and the possibilties in each category can be different in the national annex of each country and are also subject to change in the future versions.

Since other classes will depend on these classes, there should be an abstract exposure class, abstract version of each exposure class and an abstract version of ExposureClasses data container.

This was initially not done, because inheriting from Enum and ABC at the same time is not possible. Here we will apply this stackoverflow solution:
https://stackoverflow.com/questions/56131308/create-an-abstract-enum-class/56135108#56135108 

### Acceptance Criteria

- This feature should be non-breaking

### Dependencies

_No response_

### Confirmation

- [X] I have checked that this feature does not already exist in the issue tracker.
- [X] I have read and understood the contribution guidelines.

### PR DESCRIPTION
**Pull request by SZeltaat**
## Description

Refactored and replaced exposure classes. Main change is the definition of ABCEnumMeta class to be able to make Exposure an abstract class. This was required if we want to define the concept of exposure class independant of the exact definition in different Eurocode versions.
I had to do this in order to be able to do the same for the structural class and the cover calculation.
All the other changes are mostly replacing code.

Please also read #305
Fixes #305 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_4_durability_and_cover/test_table_4_1.py
Comment (egarciamendez) [inline tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_4_durability_and_cover/test_table_4_1.py:113]: To solve the errors regarding 'Statement seems to have no effect'. You could add an underscore as a variable
afbeelding

**Review by egarciamendez — APPROVED** [review]
Beautiful !
afbeelding

File: blueprints/codes/eurocode/exposure_classes.py
Comment (SZeltaat) [reply blueprints/codes/eurocode/exposure_classes.py:49]: Somehow, `other: T` works fine here and mypy doesn't mind that. For the sake of consistency, I will NOT change this one as well.

File: tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_4_durability_and_cover/test_table_4_1.py
Comment (SZeltaat) [reply tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_4_durability_and_cover/test_table_4_1.py:113]: Done!
I did not get these errors in my IDE though! 🤷‍♂️

**Review by SZeltaat — COMMENTED** [review]
Thank you for your review!
I tried changing 'object' to 'T', but mypy complains that the signature is different from its superclass. I have tried a couple of other possible solutions, but with no luck. So I'm keeping 'object' in. If you can think of a solution, please let me know or feel free to apply it yourself!
I'll wait for your response before merging! 😉

**Comment by egarciamendez** [comment]
Sometimes is a 9 just fine 👌

**Pull request merged**
